src/components/Habits/HabitForm.tsx

```diff
@@ -0,0 +1,170 @@
+
+const habitFormSchema = z.object({
+  name: z.string().min(1, { message: "Name is required" }),
+  notificationTime: z.enum(["sunrise", "sunset", "both"]),
+  emailNotificationEnabled: z.boolean(),
+  pushNotificationEnabled: z.boolean(),
+});
+
+type HabitFormValues = z.infer<typeof habitFormSchema>;
+
+type HabitFormProps = NativeStackScreenProps<HabitStackParamList, "HabitForm">;
+
+export default function HabitForm({ navigation }: HabitFormProps) {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<HabitFormValues>({
+    resolver: zodResolver(habitFormSchema),
+    defaultValues: {
+      notificationTime: "sunrise",
+      emailNotificationEnabled: true,
+      pushNotificationEnabled: true,
+    },
+  });
+
+  const { addHabit } = useHabit();
+
+  const user = useUser();
+
+  if (!user) {
+    return <Text>Please log in to add a habit</Text>;
+  }
+
+  const onSubmit: SubmitHandler<HabitFormValues> = async (data) => {
+    const {
+      name,
+      notificationTime,
+      emailNotificationEnabled,
+      pushNotificationEnabled,
+    } = data;
+
+    addHabit({
+      id: randomUUID(),
+      userId: user.uid,
+      name,
+      notificationTime,
+      emailNotificationEnabled,
+      pushNotificationEnabled,
+    });
+
+    navigation.navigate("HabitList");
+  };
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <Text>Add a habit</Text>
+      <Controller
+        control={control}
+        name="name"
+        render={({ field: { onChange, onBlur, value } }) => (
+          <TextInput
+            autoCorrect={false}
+            style={styles.input}
+            placeholder="Name"
+            onChangeText={onChange}
+            onBlur={onBlur}
+            value={value}
+          />
+        )}
+      />
+      {errors.name && (
+        <Text style={{ color: "red" }}>{errors.name.message}</Text>
+      )}
+      <Controller
+        control={control}
+        name="notificationTime"
+        render={({ field: { onChange, value } }) => (
+          <Picker
+            selectedValue={value}
+            onValueChange={onChange}
+            style={styles.picker}
+          >
+            <Picker.Item label="Sunrise" value="sunrise" />
+            <Picker.Item label="Sunset" value="sunset" />
+            <Picker.Item label="Both" value="both" />
+          </Picker>
+        )}
+      />
+      {errors.notificationTime && (
+        <Text style={{ color: "red" }}>{errors.notificationTime.message}</Text>
+      )}
+      <Text>Email Notifications</Text>
+      <Controller
+        control={control}
+        name="emailNotificationEnabled"
+        render={({ field: { onChange, value } }) => (
+          <Switch
+            trackColor={{ false: "#767577", true: "#81b0ff" }}
+            thumbColor={value ? "#f5dd4b" : "#f4f3f4"}
+            ios_backgroundColor="#3e3e3e"
+            onValueChange={onChange}
+            value={value}
```

Comment:
Should we add a way of inputting how much before or after sunrise/sunset you want to get email/push. Something that looks like [this](https://m3.material.io/components/time-pickers/overview)